torch/csrc/api/include/torch/nn/modules/embedding.h

```diff
@@ -91,15 +43,16 @@ class Embedding : public torch::nn::ModuleHolder<EmbeddingImpl> {
  public:
   using torch::nn::ModuleHolder<EmbeddingImpl>::ModuleHolder;
 
-  static Embedding from_pretrained(const torch::Tensor& embeddings, c10::optional<EmbeddingOptions> options = c10::nullopt, bool freeze = true) {
+  static Embedding from_pretrained(const torch::Tensor& embeddings, EmbeddingOptions options = {}, bool freeze = true) {
     TORCH_CHECK(embeddings.dim() == 2, "Embeddings parameter is expected to be 2-dimensional");
-    if (options != c10::nullopt) {
-      TORCH_CHECK((*options).num_embeddings() == embeddings.size(0), "Expects options.num_embeddings to be ", embeddings.size(0) , "but found ", (*options).num_embeddings());
-      TORCH_CHECK((*options).embedding_dim() == embeddings.size(1), "Expects options.embeddings_dim to be ", embeddings.size(1) , "but found ", (*options).embedding_dim());
+    if(options.num_embeddings() && options.num_embeddings()) {
+      TORCH_CHECK(*options.num_embeddings() == embeddings.size(0), "Expects options.num_embeddings to be ", embeddings.size(0) , "but found ", *options.num_embeddings());
+      TORCH_CHECK(*options.embedding_dim() == embeddings.size(1), "Expects options.embeddings_dim to be ", embeddings.size(1) , "but found ", *options.embedding_dim());
```

Comment:
For the `from_pretrained` methods of `Embedding`, I believe we don't have to define `options.num_embeddings()` and `options.num_embeddings()`, and can just take the correct values from `embeddings.size(0)` and `embeddings.size(1)`.

Reply:
Yeah but if the user passes options with options.num_embeddings != embeddings.size(0) and(or) options.embedding_dim() != embeddings.size(1), right?

Although I do think if(options.num_embeddings() && options.num_embeddings()) should be changed to something else because if user defines only one of the two values in options then we will end up giving an error(for the other value which is not defined) which probably shouldn't happen

Reply:
Discussed offline: we probably don't need to raise error if user specifies `num_embeddings` and `embedding_dim` in the options when they call `from_pretrained`, since specifying those two parameters doesn't change the behavior of `from_pretrained`. However we should raise a warning (using `TORCH_WARN`) that `num_embeddings` and `embedding_dim` parameters in options are ignored in `from_pretrained`.